aws/config.go

```diff
@@ -53,6 +53,12 @@ type Config struct {
 	// to use based on region.
 	EndpointResolver endpoints.Resolver
 
+	// EnforceShouldRetryCheck is used in the AfterRetryHandler to always call
+	// ShouldRetry. If this is set and ShouldRetry is called, then the request's
+	// Retryable field can be either nil or set. Proper handling of the Retryable
+	// field is important when setting this field.
+	EnforceShouldRetryCheck *bool
```

Comment:
Adding a small blurb about why someone would want to enable this flag would help clarify what it is for. 

Such as something about when providing a custom retry handler and how`ShouldRetry` will be handled with and without the flag enabled.